.appveyor/appveyor_helper.bat

```diff
@@ -0,0 +1,40 @@
+@echo off
+setlocal enableDelayedExpansion
+
+rem for /f %%f in ('wmic cpu get NumberOfLogicalProcessors /value ^| find "="') do set %%f
+rem set /a "ncpus=%NumberOfLogicalProcessors%*%NUMBER_OF_PROCESSORS%"
+set /a "started=0, ended=0"
+set hasmore=1
+1>nul 2>nul del *.out *.passed
+for /f "usebackq" %%A in (`dir /on /b .appveyor ^| findstr /B /R "^build_.*\.bat$"`) do (
+    if !started! lss %NUMBER_OF_PROCESSORS% (
```

Comment:
Major part of confusion was your comments in previous commits which mentioned "wmic" being used and then the code was just using NUMBER_OF_PROCESSORS. It's good that you got rid of it.
This description is helpful. Please add to point number 5 on what happens to this batch script when one of the third party compilation fails.